### PR TITLE
Feature 2594 fix and enhance systemtest framework

### LINKED ISSUE #2497: Feature: Run selected system tests
Run a single or multiple system tests from a suite of system test framework test specification (JSON definition).

To illustrate the point. Let's assume a test specification (JSON definition) contains 5 tests:

- test A
- test B
- test C
- test E
- test F

However, one only wants to run a single test. For example, test C. It should be possible to simply tell the system test CLI which test to run: `systemtestcli --run-tests "test C" …`.

Or let's further assume, one wants to run two tests, but not all 5. This should be possible by providing a list of tests to run: `systemtestcli --run-tests "test B", "test E" …`.

### LINKED ISSUE #2593: Enhance test result when not able to launch sechub job in system test
## Situation
At the moment we have only following output
```
Failed tests:
   * 'test-remote1' : Was not able to launch SecHub job
```

## Wanted
The reason shall be clear without inspecting the log output

## Solution 
- We add parts of the exception message to the test failure description
- `maximum characters` from exception message will be : `150`

Here an example for a 401 problem:
```
 Failed tests:
   * 'test-remote1' : Was not able to launch SecHub job. Reason: Was not able to create a SecHub job for project:test-project - userCreatesNewJob call failed with: 404 - {"status":404,"error":"Not Found","message...
```

### LINKED ISSUE #2594: Reference id default handling not done for remote system tests
## Situation
When somebody defines a remote system test the reference-id must be defined and there is no auto setup to default reference id!

## Wanted
Defaults shall be automatically added, like done for local parts.

### LINKED ISSUE #2596: System test documentation does not have a reference id example
## Situation
The `system test` documentation does not have a reference id example.

## Wanted
The documentation shall show a custom reference-id so it is clear how custom reference ids can be defined.

### LINKED ISSUE #2605: Provide a `cleanup` definition and phase for tests in system test framework
## Situation
Currently we can prepare something for a test (e.g. start a vulnerable web application for testing ZAP integration). But it is not possible to cleanup the preparation (e.g. stop the test application) after the test has been done.

## Wanted
It shall be possible to define a `cleanup` phase

## Solution
Introduce a `cleanup` definition inside each test configuration and handle it at runtime.

### PR DESCRIPTION
**Pull request by de-jcup**
This PR
- closes #2596 
- closes #2594 
- closes #2593 
- closes #2497
- closes #2605

### CODE REVIEW

File: sechub-systemtest/src/test/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestEngineTest.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ void sechub_job_creation_fails() throws Exception {
         engineToTest.runTest(test, runtimeContext);
 
         /* test */
-        assertEquals("Was not able to launch SecHub job", currentTestResult.getFailure().getMessage());
+        assertEquals("Was not able to launch SecHub job. Reason: no job createable", currentTestResult.getFailure().getMessage());
```
Comment (Jeeppler) [inline sechub-systemtest/src/test/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestEngineTest.java:120]:
```suggestion
        assertEquals("Was not able to launch SecHub job. Reason: unable to create job", currentTestResult.getFailure().getMessage());
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Found some small issues.

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Please fix the issues from my previous review.

File: sechub-pds-tools/src/test/java/com/mercedesbenz/sechub/pds/tools/PDSToolsCLITest.java [hunk 2]
```diff
@@ -180,7 +181,7 @@ void systemtest_does_call_systemtest_launcher_with_correct_paramters_2() throws
         /* execute */
         cliToTest.start(new String[] { "systemtest", "--workspace-rootfolder", "/path/to/workspace", "--file", "/absolute/testfile.json",
                 "--additional-resources-folder", "./additionalResources/path", "--pds-solutions-rootfolder", "/path/to/pds-solution",
-                "--sechub-solution-rootfolder", "/path/to/sechub-solution" });
+                "--sechub-solution-rootfolder", "/path/to/sechub-solution", "--run-tests", "test1,test2" });
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-pds-tools/src/test/java/com/mercedesbenz/sechub/pds/tools/PDSToolsCLITest.java:184]: I used the standard behavior of JCommander. Means you have few ways to define listes from commandline - please look at https://www.baeldung.com/jcommander-parsing-command-line-parameters#basic-types-1. 

Inside the pds tools client, the help says:

"The (comma separated) name(s) of the tests to run. When defined, only those tests are run. When nothing defined, all system tests are executed"

means: Either the name or a list of names - comma separated

**Review by Jeeppler — COMMENTED** [review]
Good job.

**Pull request merged**